lib/tasks/webpacker/compile.rake

```diff
@@ -1,35 +1,29 @@
+require "webpacker/configuration"
 REGEX_MAP = /\A.*\.map\z/
 
 namespace :webpacker do
   desc "Compile javascript packs using webpack for production with digests"
-  task :compile => :environment do
-    dist_dir = Rails.application.config.x.webpacker[:packs_dist_dir]
-    result   = `WEBPACK_DIST_DIR=#{dist_dir} NODE_ENV=production ./bin/webpack --json`
+  task compile: ["webpacker:install:verify", :environment] do
+    webpacker_config = Webpacker::Configuration.webpacker
+    result = `NODE_ENV=production ./bin/webpack`
 
     unless $?.success?
-      puts JSON.parse(result)['errors']
+      puts JSON.parse(result)["errors"]
       exit! $?.exitstatus
     end
 
-    webpack_digests = JSON.parse(result)['assetsByChunkName'].each_with_object({}) do |(chunk, file), h|
-      h[chunk] = file.is_a?(Array) ? file.find {|f| REGEX_MAP !~ f } : file
-    end.to_json
-
-    digests_path = Rails.application.config.x.webpacker[:digests_path]
-    packs_path = Rails.root.join('public', dist_dir) || File.dirname(digests_path)
-    packs_digests_path = digests_path || Rails.root.join(packs_path, 'digests.json')
-
-    FileUtils.mkdir_p(packs_path)
-    File.open(packs_digests_path, 'w+') { |file| file.write webpack_digests }
+    packs_path = Rails.root.join("public", webpacker_config[:distDir])
+    packs_digests_path = Rails.root.join(webpacker_config[:distPath], webpacker_config[:manifestFileName])
+    webpack_digests = JSON.parse(File.read(packs_digests_path))
```

Comment:
All this work should happen in the classes we have for it. The tasks should simply be invoking methods on those classes.